src/runtime_src/xocl/core/device.cpp

```diff
@@ -730,6 +730,13 @@ get_cu_memidx() const
       for (auto& cu : get_cu_range())
         mask &= cu->get_memidx_intersect();
 
+      // If all bits are set i.e. No Banks are specified. Hence return the
+      // default bank i.e. Bank 0.
+      if (mask.all()) {
+        m_cu_memidx = 0;
+	return m_cu_memidx;
+      }
+
```

Comment:
These changes are required because when there are no Banks are specified we need to select the default bank. 
Earlier, When No Banks specified, all bits are set in the "mask" and the "**for**" loop is returning the Bank0. But now as we are traversing from the higher-order we need to hardcode to Bank 0 explicitly.

Reply:
I am not understanding what triggered this change?

Reply:
Hi @stsoe , Its again due to backward compatibility.  XRT was allocating memory in the first bank If there are no banks used. Ideally this should have not been supported earlier as well, we should have thrown a error. 
To support backward compatibility, we are doing same here. If no banks are specified, allocate in first bank.
We shouldn't break the backward compatibility at this stage. We can remove this check in early cycle of 2021.1 and ask all the app developers to remove/update their test.

@saifuddin-xilinx ,
Please add a clear comment explaining above.

Reply:
Hi @saifuddin-xilinx , As discussed , lets remove this change as there are couple of CRs depends on this change.
As we already having this discussion with @stsoe and @maxzhen , lets get a conclusion there about this change.
We can send a separate PR if we need this change.

Reply:
Remove already